modules/core/src/test/java/org/apache/ignite/internal/processors/cache/transactions/TxOnCachesStopTest.java

```diff
@@ -254,10 +345,182 @@ private void runTxOnCacheStop(TransactionConcurrency conc, TransactionIsolation
+     */
+    private List<CacheConfiguration> createCacheConfigurations() {
+        String GRP_NAME = "test-destroy-group";
+
+        List<CacheConfiguration> cacheCfgs = new ArrayList<>(50);
+
+        for (int i = 0; i < CACHE_CNT; ++i) {
+            CacheConfiguration<Integer, byte[]> c = new CacheConfiguration<>("test-cache-" + i);
+            c.setBackups(2);
+            c.setAtomicityMode(CacheAtomicityMode.TRANSACTIONAL);
+            c.setWriteSynchronizationMode(CacheWriteSynchronizationMode.FULL_SYNC);
+            c.setAffinity(new RendezvousAffinityFunction(false, 32));
+            c.setGroupName(GRP_NAME);
+
+            cacheCfgs.add(c);
+        }
+
+        return cacheCfgs;
+    }
+
+    /**
+     * @param clients Client nodes that are used for initiating transactions.
+     * @param conc Transaction concurrency mode.
+     * @param iso Transaction isolation.
+     * @throws Exception If failed.
+     */
+    private void testConcurrentTransactionsOnCacheDestroy(
+        final ArrayList<Ignite> clients,
+        TransactionConcurrency conc,
+        TransactionIsolation iso
+    ) throws Exception {
+        if (log.isInfoEnabled()) {
+            log.info("Starting testConcurrentTransactionsOnCacheDestroy " +
+                "[concurrency=" + conc + ", isolation=" + iso + ']');
+        }
+
+        final AtomicBoolean stopTxLoad = new AtomicBoolean();
+        final AtomicInteger cacheIdxToBeDestroyed = new AtomicInteger(-1);
+
+        IgniteInternalFuture txLoadFut = startTxLoad(stopTxLoad, cacheIdxToBeDestroyed, clients, conc, iso);
+
+        try {
+            for (int i = 0; i < CACHE_CNT; ++i) {
+                int clientIdx = (i % clients.size());
+
+                IgniteInternalFuture destFut = GridTestUtils.runAsync(() ->
+                    clients.get(clientIdx).destroyCache("test-cache-" + cacheIdxToBeDestroyed.incrementAndGet())
+                );
+
+                try {
+                    destFut.get(15, TimeUnit.SECONDS);
+                }
+                catch (IgniteCheckedException e) {
+                    fail("Looks like PME hangs [err=" + e + ']');
+                }
+            }
+        }
+        catch (Throwable t) {
+            fail("Unexpected error [err=" + t + ']');
+        }
+
+        stopTxLoad.set(true);
+
+        txLoadFut.get();
+    }
+
+    /**
+     * Starts transactional load.
+     *
+     * @param stopTxLoad Boolean flag that is used to stop transactional load.
+     * @param cacheIdxToBeDestroyed Variable that allows to get an index of destroyed cache.
+     * @param clients Client nodes that are used for initiating transactions.
+     * @param concurrency Transaction concurrency mode.
+     * @param isolation Transaction isolation.
+     * @return TxLoad future.
+     */
+    private IgniteInternalFuture startTxLoad (
+        final AtomicBoolean stopTxLoad,
+        final AtomicInteger cacheIdxToBeDestroyed,
+        final ArrayList<Ignite> clients,
```

Comment:
ArrayList? Maybe, just list? What do you think?

Reply:
Fixed